### DISCUSSION TITLE
How can I know if the Markdown View rendered over?

### DISCUSSION THREAD

**Discussion by RHxW** [gonzalezreal/swift-markdown-ui#223] [Q&A] [answered]
Is there some way to let the caller know when the Markdown view is completely loaded?

**Comment by gonzalezreal** [comment, marked as answer]
Hi @RHxW,

There is currently no API in MarkdownUI to do that. But, since the only thing that the Markdown view loads asynchronously are images, you could replace the image-loading system with your own and track when the images are done loading.

See the [`markdownImageProvider(_:)`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.1.0/documentation/markdownui/markdown/markdownimageprovider(_:)) and [`markdownInlineImageProvider(_:)`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.1.0/documentation/markdownui/markdown/markdowninlineimageprovider(_:)) modifiers.

**Reply by RHxW** [reply]
Thanks!
I'm new to SwiftUI, your solution may be a little bit hard for me, but thanks a lot  :D

I run into this question when trying to take a screenshot using the `ImageRenderer` of SwiftUI. The problem is that the picture I took is kinda weird, it appeared in the middle of the content.
For example if I rendered a `Markdown` view with a long content(more than one screen), the screenshot will be the middle of the content, which I want it to be at the beginning/top of it. Normally the `Markdown` view displays from the top of the content.

I thought the reason might be the loading process hasn't complete yet, because if you render a `Markdown` view of long content, you can actually see the rendering process: middle -> top

So I thought maybe if I could know when does the `Markdown` view rendered over, I could tell the `ImageRenderer` when to take the screenshot.

I saw your answer in another Q&A, you said the limitation of `ImageRenderer`, it can't work with async view or `ScrollView`. 
I was confused by this a very long time
 Your answer helps me a lot ! Thanks!
